core/deployment/src/main/java/io/quarkus/deployment/dev/testing/JunitTestRunner.java

```diff
@@ -673,6 +682,24 @@ public String apply(Class<?> aClass) {
                 }
             }
         }
+        Set<DotName> processed = new HashSet<>();
+        processed.addAll(ret);
+        for (ClassInfo clazz : index.getKnownClasses()) {
+            for (DotName annotation : clazz.annotations().keySet()) {
```

Comment:
Isn't loading _all_ (used) annotations a bit expensive? OTOH I have no suggestion how to improve that (except for checking only for specific annotations).

Reply:
Not really, it is only loading a single instance for each annotation type, and already skips JUnit ones. I would not expect tests to be using a large amount of distinct annotations.